### ISSUE TITLE
Is there some way to generate responses as non-optional even if not 'required' in the schema?

### ISSUE THREAD

**Issue by kornysietsma** [drwpow/openapi-typescript#1570]
**Description**

I have a number of APIs that I need to talk to internally - however they all have openapi schemas generated using a library (C#'s Swashbuckle) which does not mark fields as required, even if they are not nullable in the original types.

This means every field in every response I get is optional:

```typescript
primary?: boolean;
```

even nullable fields are both nullable and optional:

```typescript
applicantId?: string | null;
```

I'm aware that the OpenAPI spec says the response fields need to be `required` explicitly - but it'd be nice to be able to override this where people aren't following the spec correctly.

**Proposal**

add a `--response-fields-non-optional` CLI parameter to allow all response fields to be treated as non-optional

**Checklist**

- [ ] I’m willing to open a PR for this (see [CONTRIBUTING.md](https://github.com/drwpow/openapi-typescript/blob/main/packages/openapi-typescript/CONTRIBUTING.md))

**Comment by drwpow** [comment]
I’d support adding this flag to the project. While normally I encourage just updating the original OpenAPI schema, “optional by default” responses has always been a sticking point, and probably not an intentional decision made in many schemas. In other words, a good flag for openapi-typescript either clarifies ambiguity or allows for different interpretations, rather than outright transforming something explicitly _A_ into _B_. I think there’s ambiguity around this specific behavior of the spec.

**Comment by TzviPM** [comment]
Would you be open to a PR for this?

**Comment by TzviPM** [comment]
@drwpow I submitted a PR that implements this functionality. More details on my approach are in the PR
